packages/database/migrations/20240601050128_currency_code_transaction_amm_changes/migration.sql

```diff
@@ -0,0 +1,50 @@
+/*
+  Warnings:
+
+  - You are about to drop the column `currencyId` on the `TransactionItem` table. All the data in the column will be lost.
+  - A unique constraint covering the columns `[ammId]` on the table `Market` will be added. If there are existing duplicate values, this will fail.
+  - Changed the type of `code` on the `Currency` table. No cast exists, the column would be dropped and recreated, which cannot be done if there is data, since the column is required.
+  - Added the required column `ammId` to the `Market` table without a default value. This is not possible if the table is not empty.
+  - Added the required column `creatorId` to the `Transaction` table without a default value. This is not possible if the table is not empty.
+  - Added the required column `type` to the `Transaction` table without a default value. This is not possible if the table is not empty.
+  - Added the required column `currencyCode` to the `TransactionItem` table without a default value. This is not possible if the table is not empty.
+
+*/
+-- CreateEnum
+CREATE TYPE "CurrencyCode" AS ENUM ('PRIMARY', 'YES', 'NO', 'LPB');
+
+-- DropForeignKey
+ALTER TABLE "TransactionItem" DROP CONSTRAINT "TransactionItem_currencyId_fkey";
+
+-- AlterTable
+ALTER TABLE "Currency" DROP COLUMN "code",
+ADD COLUMN     "code" "CurrencyCode" NOT NULL;
+
+-- AlterTable
+ALTER TABLE "Market" ADD COLUMN     "ammId" TEXT NOT NULL;
+
+-- AlterTable
+ALTER TABLE "Transaction" ADD COLUMN     "creatorId" TEXT NOT NULL,
+ADD COLUMN     "type" TEXT NOT NULL;
+
+-- AlterTable
+ALTER TABLE "TransactionItem" DROP COLUMN "currencyId",
+ADD COLUMN     "currencyCode" "CurrencyCode" NOT NULL;
+
+-- AlterTable
+ALTER TABLE "User" ADD COLUMN     "isAMM" BOOLEAN NOT NULL DEFAULT false;
```

Comment:
I'm not sure if it makes sense to have AMM users, since users are tied up in things like having sessions. I'd maybe go with having an "Account" table, where every User has (at least) one Account but there can also be Accounts without a User (like an AMM).

We should be able to refactor this to work like that in the future though, so not a big deal.

Also for the "Exchanger" we might want to have an "isExchanger" flag as well?

Reply:
Alternatively, maybe we just need an "isSystem" flag or something like that, to cover future cases along with the exchanger.

Reply:
Ok, yeah that makes a ton of sense. I think setting up an Account for this makes the most sense, I will adjust!

Reply:
Nice, yeah I ended up migrating the auth "Account" model that the auth library set up so that we could use the model name Account for the purposes of tying users and systems to transactions like you mentioned!